### PR TITLE
Improve development instructions

### PR DESCRIPTION
**Pull request by Fitzsimmons**
* Adds some guidance on which version of node and npm to use
* Explains how to configure the bot
* Explains how to invite the bot to a server
* Vagrant will now automatically bring up the service when the vm starts
* Provides a docker-compose.yml for administering the docker service, and prevents the image from including config.json (which is secret). config.json is automatically inserted by docker-compose instead.

Note that Docker and Vagrant execution methods will not be able to find the commands directory until #50 is merged.

### CODE REVIEW

File: vagrant-bootstrap.sh [hunk 1]
```diff
@@ -7,17 +7,15 @@ sudo curl -sL https://deb.nodesource.com/setup_7.x | sudo -E bash -
 sudo apt-get install -y nodejs
```
Comment (weylin) [inline vagrant-bootstrap.sh]: While you're here, mind bumping `setup_7.x` to `setup_8.x`?

Comment (weylin) [inline vagrant-bootstrap.sh]: That actually seems to be breaking now though... digging a little bit more.

Comment (weylin) [inline vagrant-bootstrap.sh]: Seems I had cruft that was breaking this locally, 7->8 for sure. 
Should up npm to `5.0.3` and node to `v8.1.2`

File: package.json [hunk 2]
```diff
@@ -30,6 +30,7 @@
   ],
   "dependencies": {
     "cheerio": "^0.22.0",
+    "discord.js": "^11.1.0",
```
Comment (katagatame) [inline package.json]: YAMDBF includes this, we don't need to specify a version.  Currently, it seems YAMDBF is on 11.2.0 of Discord.js.

File: README.md [hunk 3]
```diff
@@ -6,43 +6,63 @@
```
Comment (katagatame) [inline README.md]: We decided to make the README.md a bit more user facing.

I've created a branch to remove instructions from README and generated some intro text.  Your updated instructions have been migrated to the [Wiki - Installation](https://github.com/r-DestinyTheGame/Sweeper-Bot/wiki/Installation) page.

**Comment by katagatame** [comment]
I think if we trim this PR down to just the following:
  - .dockerignore
  - Dockerfile
  - docker-compose.yml
  - vagrant-bootstrap.sh (and address Weylin's concerns)

it will be good to merge.

**Comment by katagatame** [comment]
This PR is now out of date.

**Pull request closed without merging**